treebrowser/src/treebrowser.c

```diff
@@ -414,16 +420,35 @@ static gboolean
 treebrowser_checkdir(gchar *directory)
 {
 	gboolean is_dir;
-	static const GdkColor red 	= {0, 0xffff, 0x6666, 0x6666};
-	static const GdkColor white = {0, 0xffff, 0xffff, 0xffff};
 	static gboolean old_value = TRUE;
+    GtkCssProvider *provider;
+    GdkDisplay *display;
+    GdkScreen *screen;
 
 	is_dir = g_file_test(directory, G_FILE_TEST_IS_DIR);
-
+    
 	if (old_value != is_dir)
 	{
-		gtk_widget_modify_base(GTK_WIDGET(addressbar), GTK_STATE_NORMAL, is_dir ? NULL : &red);
-		gtk_widget_modify_text(GTK_WIDGET(addressbar), GTK_STATE_NORMAL, is_dir ? NULL : &white);
+        provider = gtk_css_provider_new ();
+        display = gdk_display_get_default ();
+        screen = gdk_display_get_default_screen (display);
+        gtk_style_context_add_provider_for_screen (screen, GTK_STYLE_PROVIDER (provider),
+                                                   GTK_STYLE_PROVIDER_PRIORITY_APPLICATION);
+        if(!is_dir)
+        {
+            gtk_css_provider_load_from_data (GTK_CSS_PROVIDER(provider),
+                                             " #addrbar {\n"
+                                             "   background: red;\n"
+                                             "}\n", -1, NULL);
+        }
+        else
+        {
+            gtk_css_provider_load_from_data (GTK_CSS_PROVIDER(provider),
+                                             " #addrbar {\n"
+                                             "   background: white;\n"
+                                             "}\n", -1, NULL);
+        }
```

Comment:
and actually this isn't a proper port of what the original code did in GTK2: it used to set *both* background and foreground when `is_dir` evaluated to false, and here you change only the background, but to a different color depending on the `is_dir` value.  That's not good, as you change background but not foreground, so it might very well be unreadable (imagine white on white for a dark theme).